bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingManagerImpl.java

```diff
@@ -1116,6 +1144,40 @@ private void setThingStatus(Thing thing, ThingStatusInfo thingStatusInfo) {
         }
     }
 
+    @Override
+    public void setEnabled(ThingUID thingUID, boolean enabled) {
+        Thing thing = getThing(thingUID);
+
+        if (enabled) {
+            // Enable a thing
+            logger.debug("Thing {} will be enabled.", thing.getUID().getAsString());
+            if (storage != null) {
+                storage.remove(thingUID.getAsString());
+            }
+            if (!thing.getStatus().equals(ThingStatus.ONLINE)) {
+                setThingStatus(thing, buildStatusInfo(ThingStatus.UNINITIALIZED, ThingStatusDetail.NONE));
+                registerAndInitializeHandler(thing, findThingHandlerFactory(thing.getThingTypeUID()));
+            }
+        } else {
+            logger.debug("Thing {} will be disabled.", thing.getUID().getAsString());
+            if (storage != null) {
+                storage.put(thingUID.getAsString(), enabled);
+            }
+            ThingHandlerFactory thingHandlerFactory = findThingHandlerFactory(thing.getThingTypeUID());
+            if (thing.getHandler() != null) {
```

Comment:
You have created a funny little asymmetry here: While `unregisterAndDisposeHandler(...)` also unregisters and disposes all child handlers in case it is a bridge, your `else` block enters the game at a place where this does not happen anymore. This leads to quite surprising side-effects, I reckon.

Reply:
The for checking the status of the thing is moved to ```registerAndInitializeHandler(...)``` method. This  also covers ```thingUpdated(...)``` entry point.